app/main/api/handlers/newMnemonics.ts

```diff
@@ -10,11 +17,85 @@ import * as mnemonic from "app/main/crypto/mnemonic"
  * @param params
  */
 export default async function newMnemonics(system: AppStateS, params: any) {
-  // Get mnemonics from crypto library
-  const mnemonicsResponse: NewMnemonicsResponse = { mnemonics: mnemonic.generate() }
+  return Promise.resolve()
+    // Get mnemonics from crypto library
+    .then(generateMnemonics)
+    // Update app state with unconsolidated wallet entry
+    .then(inject(updateUnconsolidatedWallet, system))
+    // create response steps
+    .then(buildSuccess)
+    .catch(buildError)
+    .then(encodeResponse)
+}
+
+/**
+ * Get mnemonics from crypto library
+ *
+ * @returns {Mnemonics}
+ */
+function generateMnemonics(): Mnemonics {
+  try {
+    return { mnemonics: mnemonic.generate() }
+  } catch (err) {
+    throw newMnemonicsErrors.DEPENDENCY_ERROR_GENERATE_MNEMONICS
+  }
+}
 
-  // Update app state with unconsolidated wallet entry
-  system.appStateManager.update({ unconsolidatedWallet: mnemonicsResponse })
+/**
+ * update app state with unconsolidated wallet entry
+ *
+ * @param {Mnemonics} mnemonics
+ * @param {AppStateS} system
+ * @returns {Mnemonics}
+ */
+function updateUnconsolidatedWallet(
+  mnemonics: Mnemonics, system: AppStateS
+): Mnemonics {
```

Comment:
I kind of prefer putting the "classic js" signature in a single line and moving the return type to the next one if it doesn't fit in the max character count per line:
```ts
 function updateUnconsolidatedWallet(mnemonics: Mnemonics, system: AppStateS)
  : Mnemonics {
```